File: rails/rate.rb
require 'open-uri'

def get_rate(cur_code)
  url = "https://api.paybear.io/v1/#{cur_code.downcase!}/exchange/usd/rate"

  response = ActiveSupport::JSON.decode(open(url).read)

  if response['success']
    response['data']['mid']
  end
end

File: php/rate.php
<?php
function getRate($curCode) {
    $url = "https://api.paybear.io/v1/".strtolower($curCode)."/exchange/usd/rate";

    if ($response = file_get_contents($url)) {
        $response = json_decode($response);
        if ($response->success) {
            return $response->data->mid;
        }
    }
    return null;
}

